packages/components/src/components/step/step.ts

```diff
@@ -140,21 +141,23 @@ export default class SdStep extends SolidElement {
 
           <${tag}
             part="circle"
-            ?disabled=${this.disabled}
-            tabindex=${!this.disabled && !this.current ? '0' : '-1'}
             href=${ifDefined(isLink ? this.href : undefined)}
+            aria-disabled=${ifDefined(this.disabled || undefined)}
             aria-current=${this.current ? 'step' : undefined}
-            aria-label='Step ${this.index}'
+            aria-labelledby=${ifDefined('label')}
+            aria-describedby=${ifDefined('description')}
             class=${cx(
               'border rounded-full aspect-square circle flex items-center justify-center shrink-0 font-bold select-none',
-              !this.disabled && !this.current ? 'focus-visible:focus-outline' : 'focus-visible:outline-none',
+              !this.disabled
+                ? 'focus-visible:focus-outline group-hover:cursor-pointer '
+                : 'focus-visible:outline-none cursor-not-allowed',
```

Comment:
It is easier to read if the expression is positive instead of negative.
[Avoid Negative Conditionals](https://www.franciscomoretti.com/code-tips/avoid-negative-conditionals)

```suggestion
              this.disabled
                ? 'focus-visible:outline-none cursor-not-allowed'
                : 'focus-visible:focus-outline group-hover:cursor-pointer',
```